tests/regression/halmos.toml

```diff
@@ -0,0 +1,148 @@
+
+# set the byte size of return data from uninterpreted unknown external calls
+return-size-of-unknown-calls = 32
+
+# Select one of the available storage layout models
+# The generic model should only be necessary for vyper, huff, or unconventional storage patterns in yul.
+storage-layout = "solidity"
+
+# set default storage values to symbolic
+symbolic-storage = false
+
+# set msg.sender symbolic
+symbolic-msg-sender = false
+
+# do not run the constructor of test contracts
+no-test-constructor = false
+
+# allow the usage of FFI to call external functions
+ffi = false
+
+################################################################################
+#                              Debugging options                               #
+################################################################################
+
+# increase verbosity levels: -v, -vv, -vvv, ...
+verbose = 0
+
+# print statistics
+statistics = false
+
+# run in debug mode
+debug = false
+
+# log every execution steps in JSON
+# log = LOG_FILE_PATH
+
+# output test results in JSON
+# json-output = JSON_FILE_PATH
+
+# include minimal information in the JSON output
+minimal-json-output = false
+
+# print every execution step
+print-steps = false
+
+# when --print-steps is enabled, also print memory contents
+print-mem = false
+
+# print all final execution states
+print-states = false
+
+# print failed execution states
+print-failed-states = false
+
+# print blocked execution states
+print-blocked-states = false
+
+# print setup execution states
+print-setup-states = false
+
+# print full counterexample model
+print-full-model = false
+
+# stop after a counterexample is found
+early-exit = false
+
+# dump SMT queries for assertion violations
+dump-smt-queries = false
+
+################################################################################
+#                                Build options                                 #
+################################################################################
+
+# forge build artifacts directory name
+forge-build-out = "out"
+
+################################################################################
+#                                Solver options                                #
+################################################################################
+
+# interpret constant power up to N
+smt-exp-by-const = 2
+
+# set timeout (in milliseconds) for solving branching conditions; 0 means no timeout
+solver-timeout-branching = 1
+
+# set timeout (in milliseconds) for solving assertion violation conditions; 0 means no timeout
+solver-timeout-assertion = 1000
+
+# set memory limit (in megabytes) for the solver; 0 means no limit
+solver-max-memory = 0
+
+# use the given command when invoking the solver
+# solver-command = COMMAND
+
+# run assertion solvers in parallel
+solver-parallel = false
+
+# set the number of threads for parallel solvers
+solver-threads = 20
```

Comment:
```suggestion
# solver-threads = N
```

Reply:
the value for --solver-threads doesn't need to be set here, as it can be set at runtime based on the runner's cpu info.

can we have the auto-generation to set solver-threads only when it is explicitly given by command line?

Reply:
yes you're right